wordpress-mobile/WordPress-Android
# Bump Gravatar SDK to 2.2.0

This PR bumps the Gravatar SDK to 2.2.0. You can see the release notes [here](https://github.com/Automattic/Gravatar-SDK-Android/releases/tag/2.2.0).

In short, this updates brings two new options to manage your existing Avatars:
- Delete
- Download

There were other fixes and updates, but these two are the most noticeable ones for the WP app.

On the Dependency side of things. We no longer use `moshi-kotlin` which means `kotlin-reflect` is not longer added. That will mean a smaller app size.

## To Test:

Technically speaking, there's not much to test on the WP side since the features provided in the SDK have already been tested. Feel free to test this yourself. 

1. Go to the MeFragment by tapping the Me option
2. Tap on the avatar
3. Confirm the QuickEditor was opened
4. Upload a new avatar if you don't have any
5. Tap on the `...` button on an Avatar and pick one of the options

You can 

## Regression Notes

You can see that I had to add `tools:replace="android:maxSdkVersion"` to the `AndroidManifest.xml`. This ensures that no changes to the WP manifest are done and the `maxSdkVersion=29` is kept. Quick Editor needs the `android.permission.WRITE_EXTERNAL_STORAGE` permission but only up to API 28, so that's where the conflict is. This permission is needed for the DownloadManager (see [docs](https://developer.android.com/reference/android/app/DownloadManager.Request#setDestinationInExternalPublicDir(java.lang.String,%20java.lang.String))) to download Avatars. 

Here's a screenshot of the merge AndroidManifest from the .apk confirming no changes are made. 

<img width="519" alt="Screenshot 2024-12-18 at 11 53 40" src="https://github.com/user-attachments/assets/8def4035-d60e-474b-b237-99235b5e2074" />

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
